WordPress/Classes/ViewRelated/Blog/CreateNewBlog/CreateNewBlogViewController.m

```diff
@@ -0,0 +1,551 @@
+#import "Constants.h"
+
+#import "WordPress-Swift.h"
+
+@interface CreateNewBlogViewController ()<UITextFieldDelegate,UIGestureRecognizerDelegate> {
+    WPNUXSecondaryButton *_cancelButton;
+    UILabel *_titleLabel;
+    UILabel *_siteAddressWPComLabel;
+    WPWalkthroughTextField *_siteTitleField;
+    WPNUXMainButton *_createBlogButton;
+    WPWalkthroughTextField *_siteAddressField;
+    
+    NSOperationQueue *_operationQueue;
+    
+    BOOL _authenticating;
+    BOOL _keyboardVisible;
+    BOOL _userDefinedSiteAddress;
+    CGFloat _keyboardOffset;
+    
+    NSDictionary *_currentLanguage;
+}
+
+@end
+
+@implementation CreateNewBlogViewController
+
+static CGFloat const CreateBlogStandardOffset             = 15.0;
+static CGFloat const CreateBlogTextFieldWidth             = 320.0;
+static CGFloat const CreateBlogTextFieldHeight            = 44.0;
+static CGFloat const CreateBlogTextFieldPhoneHeight       = 38.0;
+static CGFloat const CreateBlogiOS7StatusBarOffset        = 20.0;
+static CGFloat const CreateBlogButtonWidth                = 290.0;
+static CGFloat const CreateBlogButtonHeight               = 41.0;
+
+static UIEdgeInsets const CreateBlogCancelButtonPadding     = {1.0, 0.0, 0.0, 0.0};
+static UIEdgeInsets const CreateBlogCancelButtonPaddingPad  = {1.0, 13.0, 0.0, 0.0};
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _operationQueue = [[NSOperationQueue alloc] init];
+        _currentLanguage = [WPComLanguages currentLanguage];
+    }
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.view.backgroundColor = [WPStyleGuide wordPressBlue];
+    
+    [self initializeView];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:)
+                                                 name:UIKeyboardWillShowNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillHide:)
+                                                 name:UIKeyboardWillHideNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardDidShow)
+                                                 name:UIKeyboardDidShowNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardDidHide)
+                                                 name:UIKeyboardDidHideNotification object:nil];
+}
+
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+    [self.navigationController setNavigationBarHidden:YES animated:animated];
+    [self layoutControls];
+}
+
+- (UIInterfaceOrientationMask)supportedInterfaceOrientations
+{
+    if (IS_IPHONE) {
+        return UIInterfaceOrientationMaskPortrait;
+    }
+    
+    return UIInterfaceOrientationMaskAll;
+}
+
+- (void)willAnimateRotationToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation
```

Comment:
@kwonye this is missing a `super` call, but it is deprecated anyways: `"Implement viewWillTransitionToSize:withTransitionCoordinator: instead"`